### PR TITLE
Use invoice lines' analytic account for block hours total

### PR DESCRIPTION
**Pull request by leemannd**
Add changes when using multiple hours block for different analytic accounts.

### CODE REVIEW

**Comment by mdietrichc2c** [comment]
Not sure why the Travis failed ; in any case, :+1: on this PR as it is unrelated.

File: analytic_hours_block/hours_block.py [hunk 1]
```diff
@@ -50,7 +50,11 @@ def _compute_hours(self, cr, uid, ids, fields, args, context=None):
             # Compute hours bought
             for line in block.invoice_id.invoice_line:
                 hours_bought = 0.0
-                if line.product_id and line.product_id.is_in_hours_block:
+                block_analytic_id = block.account_analytic_id.id
```
Comment (vrenaville) [inline analytic_hours_block/hours_block.py]: We can move this assignation outside of the for (line 50)

**Comment by vrenaville** [comment]
except my comment :+1:

**Comment by dreispt** [comment]
I'm marking WIP until CLA issue is resolved.

**Comment by dreispt** [comment]
Is this related to #89?

**Comment by leemannd** [comment]
@dreispt It's not related to https://github.com/OCA/project/pull/89
The hours block didn't take the related analytical account.
I have signed and submitted the CLA

**Comment by dreispt** [comment]
Thank you @leemannd. TravisCI error is unrelated, so I'm marking as "Need Review".

File: analytic_hours_block/hours_block.py [hunk 2]
```diff
@@ -173,6 +177,16 @@ def _get_invoice(self, cr, uid, ids, context=None):
                               in invoice.account_hours_block_ids])
         return list(block_ids)
 
+    def _get_invoice_line(self, cr, uid, ids, context=None):
+        invoice_ids = set()
+        line_obj = self.pool.get('account.invoice.line')
+        block_obj = self.pool.get('account.hours.block')
```
Comment (guewen) [inline analytic_hours_block/hours_block.py]: This form is better: `block_obj = self.pool['account.hours.block']`.
Explanation:
When `account.hours.block` is not in `self.pool` for some reason:
- `self.pool.get('account.hours.block')` will return None, so when later you do `block_obj._get_invoice(...)`, will raise an AttributeError `'NoneType' object has no attribute '_get_invoice'` which is hard to debug
- `self.pool['account.hours.block']` will fail directly, which makes it easier to find the issue

File: analytic_hours_block/hours_block.py [hunk 3]
```diff
@@ -47,10 +47,14 @@ def _compute_hours(self, cr, uid, ids, fields, args, context=None):
...
                                 'amount_hours_block_done': 0.0}
+            block_analytic_id = block.account_analytic_id.id
             # Compute hours bought
             for line in block.invoice_id.invoice_line:
                 hours_bought = 0.0
-                if line.product_id and line.product_id.is_in_hours_block:
+                line_account_analytic_id = line.account_analytic_id.id
+                if line.product_id and \
+                        line.product_id.is_in_hours_block and \
+                        block_analytic_id == line_account_analytic_id:
```
Comment (guewen) [inline analytic_hours_block/hours_block.py]: `\` for continuation lines are to be avoided, parentheses should be preferred. Example:

``` python
if (line.product_id and
        line.product_id.is_in_hours_block and
        block_analytic_id == line_account_analytic_id):
```

**Comment by leemannd** [comment]
@guewen Correction made. Thank you.

**Comment by guewen** [comment]
Thanks for the changes! :+1:

**Pull request merged**